lua-mosquitto.c

```diff
@@ -74,16 +71,26 @@ typedef struct {
 	lua_State *L;
 	struct mosquitto *mosq;
 	int on_connect;
+	int on_connect_v5;
 	int on_disconnect;
+	int on_disconnect_v5;
 	int on_publish;
+	int on_publish_v5;
 	int on_message;
+	int on_message_v5;
 	int on_subscribe;
+	int on_subscribe_v5;
 	int on_unsubscribe;
+	int on_unsubscribe_v5;
```

Comment:
All the callbacks can stay the same right? you can't have v3 and v5 in the same client object, so there's no confusion there.

Reply:
libmosquitto will call both callbacks when i.e. `mosquitto_connect_callback_set` and `mosquitto_connect_v5_callback_set` have been called before.

Reply:
is that actually meaningful?  a client can only be one protocol at a time, so being able to register two, and getting two callbacks just seems like complexity for nothing

Reply:
I think the idea for the v5 api design of luamosquitto is, that old code can stay unmodified and meanwhile can be extended with v5 functionality by just adding postfixed `_v5` functions to your codebase.

Now since Lua is much more flexible than C, because of optional arguments and so on, there could probably also be a method to reach the same goal with just extending the existing API. But this would mean, that from there on this lib is more than just a transparent wrapper for libmosquitto and would probably involve some additional documentation work since you can not simply refer to the `mosquitto.h` and say this is our API.

I would say the biggest work in this PR was the properties parsing back and forth so changing the design decision on this point might not be catastrophic. 

I like the current approach more because it is way more explicit but I am open to your final decision :)